R-packages/modeltools/R/response.R

```diff
@@ -0,0 +1,80 @@
+library(assertthat)
+
+#' Compute the corresponding observed response values for a set of signals.
+#'
+#' The response for an observed data point at time `t` is defined as the mean of observed values on
+#' days `t + s` through `t + e` (inclusive) multiplied by `e - s`, where `s` and `e` are the number
+#' of days from `forecast_date` to the start and end, respectively, of the next full
+#' `incidence_period`.  We choose to take the mean and multiply by the incidence period length
+#' rather than simply sum the observed values in order to avoid treating missing values as 0.
+#'
+#' @param signals Wide data frame of signals.  We expect the following columns:
+#'     \itemize{
+#'     \item{`geo_value`}{Strings of geographic locations.}
+#'     \item{`time_value`}{Dates of training data.}
+#'     \item{Covariate columns}{Columns with names of the form `value-{days}:{signal}` or
+#'         `value+0:{signal} whose values correspond to `{signal}` `{days}` before `time_value`}
+#'     }
+#' @param response_name Name of response signal.  The value of the response variable on a single
+#'     day should be located in column `value+0:{response_name}` of `signals`.
+#' @param forecast_date Date on which the forecast will be made
+#' @param incidence_period Time period over which the response should be summed.
+#' @param ahead integer or vector of integer ahead values
+#'
+#' @return Data frame of signals equal to `signals` with `length(ahead)` additional columns with
+#'     names `response+{a}:{response_name}` corresponding to the response variable at ahead `a`. 
+#'
```

Comment:
Sorry one last comment: could we add one instance of example code?

Reply:
Done